api/src/main/java/org/openmrs/CohortMembership.java

```diff
@@ -99,6 +118,18 @@ public void setEndDate(Date endDate) {
 
 	@Override
 	public int compareTo(CohortMembership o) {
-		return this.getPatient().getPatientId() - o.getPatient().getPatientId();
+		if (Objects.equals(this.getPatient().getPatientId(), o.getPatient().getPatientId())
+				&& Objects.equals(this.getCohort().getCohortId(), o.getCohort().getCohortId())
+				&& this.getStartDate().equals(o.getStartDate())
+				&& ((this.getEndDate() != null && o.getEndDate() != null
+				&& this.getEndDate().equals(o.getEndDate()))
+				|| (this.getEndDate() == null && o.getEndDate() == null))) {
+			return 0;
+		} else if ((this.isMembershipActive() && !o.isMembershipActive())
+			|| (!this.isMembershipActive() && o.isMembershipActive())) {
```

Comment:
Understood. Since I am already comparing startDates in the first if condition, won't it taken care of already. What will be the if condition to return 1? I didn't understand from your last comment. This is what I've got:

```
		if (Objects.equals(this.getPatient().getPatientId(), o.getPatient().getPatientId())
				&& Objects.equals(this.getCohort().getCohortId(), o.getCohort().getCohortId())
				&& this.getStartDate().equals(o.getStartDate())
				&& ((this.getEndDate() != null && o.getEndDate() != null
				&& this.getEndDate().equals(o.getEndDate()))
				|| (this.getEndDate() == null && o.getEndDate() == null))) {
			return 0;
		} else if ((this.isMembershipActive() && !o.isMembershipActive())
			|| (!this.isMembershipActive() && o.isMembershipActive())) {
			return -1;
		} else if () {
			return 1;
		}
```